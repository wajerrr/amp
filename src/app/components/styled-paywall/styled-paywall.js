import React from 'react';
import { css } from 'emotion';
import { Paywall, Regwall } from '../paywall/paywall';
import color from '../../styles/color';
import fontFamily from '../../styles/font-family';
import spacings from '../../styles/spacings';
import { text } from '../../styles';
import sizes from '../../styles/icon-sizes';

const wrapperStyles = css`
  position: relative;
  box-sizing: border-box;
  padding-bottom: ${spacings.s};
  background-color: ${color.berlin};
  font-family: ${fontFamily.serif};
  margin-bottom: ${spacings.l};
`;
const sectionStyles = css`
  height: 50px;
  box-sizing: border-box;
  font-size: ${text.sizeStep['-1']};
  line-height: 1;
  text-align: center;
  background-color: ${color.thimphu};
  max-width: none;
  width: 100%;
  color: ${color.kiev};
  font-family: ${fontFamily.sans};
  & a {
    border: none;
    color: ${color.kiev};
    font-weight: bold;
    &:visited {
      color: ${color.chicago};
    }
  }
`;
const sectionContentStyles = css`
  line-height: 2;
`;
const headerStyles = css`
  margin: 0 auto;
  text-align: center;
  max-width: 80%;
  padding-top: ${spacings.xs};
`;
const headerTitleStyles = css`
  font-size: ${text.sizeStep['5']};
  line-height: ${text.lineHeight.sansOnStep['5']};
  font-weight: 500;
  margin: ${spacings.m} 0;
`;
const headerTextStyles = css`
  font-size: ${text.sizeStep['-1']};
  font-weight: 400;
  line-height: 28px;
`;
const listWrapperStyles = css`
  max-width: 85%;
  box-sizing: border-box;
  margin: 0 auto;
`;
const listStyles = css`
  padding: ${spacings.none} ${spacings.s};
`;
const listItemStyles = css`
  position: relative;
  box-sizing: border-box;
  display: block;
  font-size: ${text.sizeStep['-1']};
  line-height: 27.86px;
  padding-left: 30px;
  font-family: ${fontFamily.sans};
  letter-spacing: -0.025em;
  &:before {
    position: absolute;
    top: 4px;
    left: 0;
    content: '';
    width: ${sizes.xSmall}px;
    height: ${sizes.xSmall}px;
    background-image: url('https://i.piano.io/managedservices/theeconomist/tick-icon.png');
    background-repeat: no-repeat;
    background-size: auto;
  }
`;
const textStyles = css`
  max-width: 80%;
  margin: ${spacings.xl} auto;
  font-size: ${text.sizeStep['-1']};
`;
const imageWrapperStyles = css`
  max-width: 80%;
  margin: ${spacings.xl} auto;
`;
const subscribeButtonWrapperStyles = css`
  margin: 0 auto;
  max-width: 80%;
  & a {
    font-size: 16px;
  }
`;
const footerStyles = css`
  max-width: 80%;
  text-align: center;
  margin: ${spacings.xl} auto;
  font-family: ${fontFamily.sans};
  font-size: ${text.sizeStep['-1']};
  & a {
    border: none;
    &:visited {
      color: ${color.chicago};
    }
  }
`;

const paywallLoginTextButton = css`
  color: inherit;
  text-decoration: none;
  padding-bottom: 1px;
  border-bottom: 2px solid ${color.kosice};
  &:hover,
  &:visited {
    color: ${color.chicago};
    border-bottom-color: ${color.chicago};
  }
`;

const paywallButton = css``;

const styles = {
  section: sectionStyles,
  sectionContent: sectionContentStyles,
  paywallLoginTextButton,
  paywallButton,
  wrapper: wrapperStyles,
  header: headerStyles,
  headerTitle: headerTitleStyles,
  headerText: headerTextStyles,
  listWrapper: listWrapperStyles,
  list: listStyles,
  listItem: listItemStyles,
  text: textStyles,
  subscribeButtonWrapper: subscribeButtonWrapperStyles,
  imageWrapper: imageWrapperStyles,
  footer: footerStyles,
};

const StyledPaywall = () => <Paywall styles={styles} />;
const StyledRegwall = () => <Regwall styles={styles} />;

export { StyledPaywall, StyledRegwall };
