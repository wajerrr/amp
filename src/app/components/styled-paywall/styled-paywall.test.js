import React from 'react';
import renderer from 'react-test-renderer';
import { StyledPaywall, StyledRegwall } from './styled-paywall';

describe('Paywall and Regwall', () => {
  it('should match the snapshot for the Regwall', () => {
    const paywall = renderer.create(<StyledRegwall />).toJSON();
    expect(paywall).toMatchSnapshot();
  });
  it('should match the snapshot for the Paywall', () => {
    const paywall = renderer.create(<StyledPaywall />).toJSON();
    expect(paywall).toMatchSnapshot();
  });
});
