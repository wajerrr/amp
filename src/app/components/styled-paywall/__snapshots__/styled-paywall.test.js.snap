// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Paywall and Regwall should match the snapshot for paywall data 1`] = `
<section
  className="css-793fq4-wrapperStyles"
>
  <header
    className="css-1rpvrys-headerStyles"
  >
    <h2>
      You’ve reached your article limit
    </h2>
    <p>
      Subscribe now to get full access to 
      <i>
        The Economist
      </i>
       via print, online and our apps.
    </p>
  </header>
  <div
    className="css-mt52t7-subscribeButtonWrapperStyles"
  >
    <a
      className="e18299l00 css-18wxk9z-StyledLink-LargeSubscribeButton e1ob2h4l0"
      href="https://econ.st/2bs1gXz"
      role="button"
    >
      Subscribe
    </a>
  </div>
  <section
    className="css-ewotou-listWrapperStyles"
  >
    <ul
      className="css-1csi2et-listStyles"
    >
      <li
        className="css-igp5vx-listItemStyles"
      >
        <b>
          Print or digital edition
        </b>
         - 
        delivered to you each week
      </li>
      <li
        className="css-igp5vx-listItemStyles"
      >
        <b>
          Economist.com
        </b>
         - 
        including blog content updated throughout the week and our online archive
      </li>
      <li
        className="css-igp5vx-listItemStyles"
      >
        <b>
          Audio edition
        </b>
         - 
        each week’s issue read by professional broadcasters
      </li>
      <li
        className="css-igp5vx-listItemStyles"
      >
        <b>
          The Economist Espresso
        </b>
         - 
        your morning briefing direct to your smartphone or inbox
      </li>
    </ul>
  </section>
  <section
    className="css-1a0behp-imageWrapperStyles"
  >
    <amp-img
      alt="paywall-image"
      height={900}
      layout="responsive"
      sizes={undefined}
      src="https://i.piano.io/managedservices/theeconomist/regwall-product.png"
      srcset={undefined}
      width={1600}
    />
  </section>
  <footer
    className="css-12czfv6-footerStyles"
  >
    <p>
      Student and gift subscriptions also available.
       
      <a
        className="css-1h5t0lq-StyledLink e2cf1b50"
        href="https://subscription.economist.com/DE/EngCore/Ecom/SubWall"
      >
        Subscribe now
      </a>
    </p>
  </footer>
</section>
`;

exports[`Paywall and Regwall should match the snapshot for regwall data 1`] = `
<section
  className="css-793fq4-wrapperStyles"
>
  <section
    className="css-8n2js-sectionStyles"
  >
    <p
      className="css-hggkb-sectionContentStyles"
    >
      Already signed up or a subscriber?
       
      <a
        className="css-1h5t0lq-StyledLink e2cf1b50"
        href="https://authenticate.economist.com/login"
      >
        Log in
      </a>
    </p>
  </section>
  <header
    className="css-1rpvrys-headerStyles"
  >
    <h2>
      You’ve reached your article limit
    </h2>
    <p>
      Sign up to keep reading or subscribe now to get full access to
       
      <i>
        The Economist
      </i>
       via print, online and our apps.
    </p>
  </header>
  <div
    className="css-mt52t7-subscribeButtonWrapperStyles"
  >
    <a
      className="e18299l00 css-18wxk9z-StyledLink-LargeSubscribeButton e1ob2h4l0"
      href="https://econ.st/2bs1gXz"
      role="button"
    >
      Subscribe
    </a>
  </div>
  <p
    className="css-mvffl9-textStyles"
  >
    <i>
      The Economist
    </i>
     delivers incisive analysis on the issues that matter.
  </p>
  <div
    className="css-mt52t7-subscribeButtonWrapperStyles"
  >
    <a
      className="e18299l01 css-1aftcze-StyledLink-LargeSubscribeButton-LargeSubscribeButtonInvertedStyles e1ob2h4l0"
      href="https://econ.st/2bs1gXz"
      role="button"
    >
      Sign up: 3 articles per week
    </a>
  </div>
  <section
    className="css-1a0behp-imageWrapperStyles"
  >
    <amp-img
      alt="paywall-image"
      height={900}
      layout="responsive"
      sizes={undefined}
      src="https://i.piano.io/managedservices/theeconomist/regwall-product.png"
      srcset={undefined}
      width={1600}
    />
  </section>
  <p
    className="css-mvffl9-textStyles"
  >
    Subscribe now to enjoy:
  </p>
  <section
    className="css-ewotou-listWrapperStyles"
  >
    <ul
      className="css-1csi2et-listStyles"
    >
      <li
        className="css-igp5vx-listItemStyles"
      >
        <b>
          Print or digital edition
        </b>
         - 
        delivered to you each week
      </li>
      <li
        className="css-igp5vx-listItemStyles"
      >
        <b>
          Economist.com
        </b>
         - 
        including blog content updated throughout the week and our online archive
      </li>
      <li
        className="css-igp5vx-listItemStyles"
      >
        <b>
          Audio edition
        </b>
         - 
        each week’s issue read by professional broadcasters
      </li>
      <li
        className="css-igp5vx-listItemStyles"
      >
        <b>
          The Economist Espresso
        </b>
         - 
        your morning briefing direct to your smartphone or inbox
      </li>
    </ul>
  </section>
  <footer
    className="css-12czfv6-footerStyles"
  >
    <p>
      Student and gift subscriptions also available.
       
      <a
        className="css-1h5t0lq-StyledLink e2cf1b50"
        href="https://subscription.economist.com/DE/EngCore/Ecom/RegWall"
      >
        Subscribe now
      </a>
    </p>
  </footer>
</section>
`;
